### DISCUSSION TITLE
ArtifactGC Pod fails with ImagePull Error when ArgoExec image is in a private registry

### DISCUSSION THREAD

**Discussion by ramanNarasimhan77** [argoproj/argo-workflows#11976] [Q&A] [answered]
I have an ArgoWorkflow installation wherein all Images are being pulled from a secured private registry.  I use the community helm chart to provide PullSecrets and override Image Repositories.

When I try to run this ArtifactGC example, [https://raw.githubusercontent.com/argoproj/argo-workflows/master/examples/artifact-gc-workflow.yaml](https://raw.githubusercontent.com/argoproj/argo-workflows/master/examples/artifact-gc-workflow.yaml)

It creates a WorkflowArtifactGCTask. However pod created by this Custom Resource, is unable to Pull ArgoExec image.

I checked https://argoproj.github.io/argo-workflows/executor_swagger/#artifactgc and tried passing ServiceAccount as well in ArtifactGC section and also set the Pull Secret in serviceAccount as there is no direct provision to specify it

My observation was that serviceAccount was not set in the WorkflowArtifactGCTask or the pod created by it.

I wonder if I am doing something wrong or if this is a bug in ArgoWorkflow's ArtifactGC?

**Comment by agilgur5** [comment, marked as answer]
You should be able to configure `serviceAccountName` or `podSpecPatch` [on `artifactGC`](https://argoproj.github.io/argo-workflows/fields/#workflowlevelartifactgc) to set an `imagePullSecret`.

If you think the SA was not set correctly, then that would indeed be a bug, and you'd need to provide a Workflow that reproduces it in a bug report.
